### DISCUSSION TITLE
Using NBA player.stats

### DISCUSSION THREAD

**Discussion by kelseyfglenn** [cwendt94/espn-api#187] [ESPN API]
Curious if someone could explain what I'm seeing when calling Player.stats for the basketball API. I identified the first entry as the current matchup, but I can't figure out the time periods which the other entries cover. They seem to represent a varying number of total games and none seem to be previous weeks  or season totals and the dict keys aren't very informative so I'm a little lost.

Also, is league.player_info exclusive to the football API? If so, what would be the recommended way to access individual games or averages from previous weeks by player?

Thanks for any help, this is a great project 👍

**Comment by cwendt94** [comment]
Thats a great question @kelseyfglenn! Here are the key meanings

-  0020XX is full season stats for 20XX (e.g. 002021)
-  1020XX is projected season stats for 20XX (e.g. 102021)
-  I think 0120XX, 0220XX, and 0320XX are last 7, last 15, and last 30 (e.g. 012021)

I agree, the dict keys are not helpful and can be converted into something better. I will look into making them more readable.

I believe `player_info` can be ported over to Basketball as well but looking up individual games I'm not sure. I will have to dig through some of the API data to see how they are getting it!

**Reply by kelseyfglenn** [reply]
ah, that clears things up a lot, thank you!

on a semi-related note, do you have an idea why box_players might not be properly displaying team and player points? When looking at box_scores for a h2h cat league, I noticed that that the box_scores seem to display everything as 0, including player points. This is consistent across every matchup and scoring period (the image here is from an incomplete current period but the same happens for previous periods with a declared box_scores.winner).
![image](https://user-images.githubusercontent.com/63033406/110037957-fa1c7c00-7cf3-11eb-821d-351ca3e596ad.png)

but weirdly, the actual box_player objects inside seem to contain the full points data. trying to figure out why they're not surfacing in the box_scores.
![image](https://user-images.githubusercontent.com/63033406/110038271-5da6a980-7cf4-11eb-8ad4-931e0fe7615b.png)

**Reply by cwendt94** [reply]
Interesting, maybe category scoring is saved in a different field. Here is my output in a regular H2H league
```
box = league.box_scores(matchup_period=3, scoring_period=20)
box[0].__dict__
{
   "winner": "AWAY",
   "home_team": "Team(Team Casanova)",
   "home_projected": -1,
   "home_score": 1174.0,
   "home_lineup": [Player(Kyle Lowry, points: 119), 
      Player(Jarrett Allen, points: 157), 
      Player(Chris Paul, points: 105), 
      Player(Paul George, points: 140), 
      Player(Aaron Gordon, points: 100), 
      Player(Tyler Herro, points: 79), 
      Player(Devonte Graham, points: 119), 
      Player(Lou Williams, points: 80), 
      Player(Zion Williamson, points: 122), 
      Player(Giannis Antetokounmpo, points: 153)
   ],
   "away_team": Team(Team Fatchance),
   "away_score": 1320.0,
   "away_lineup": [Player(Ben Simmons, points: 99), 
      Player(Rui Hachimura, points: 10), 
      Player(Keldon Johnson, points: 66), 
      Player(RJ Barrett, points: 43), 
      Player(DeMar DeRozan, points: 122), 
      Player(Ricky Rubio, points: 5), 
      Player(Jonas Valanciunas, points: 85), 
      Player(Jusuf Nurkic, points: 107), 
      Player(Dillon Brooks, points: 49), 
      Player(Joel Embiid, points: 155), 
      Player(James Harden, points: 173), 
      Player(Robert Covington, points: 34), 
      Player(DeAndre Hunter, points: 43), 
      Player(Terrence Ross, points: 17),
      Player(Mike Conley, points: 122), 
      Player(Goran Dragic, points: 30), 
      Player(P.J. Washington, points: 160)
   ],
   "away_projected": -1
}
``` 
Would you be able to post the data returned from this call https://github.com/cwendt94/espn-api/blob/master/espn_api/basketball/league.py#L168. I'm not in a category league to check the data

**Reply by kelseyfglenn** [reply]
Sorry, I don't quite follow -- did you want me to call ._get_pro_schedule() for some period? 

Though, I'm now thinking it may not matter since I'm realizing that the score and points metrics for points leagues don't have any meaning in a cat league, so the box_player points and home/away_score may just be vestigial. If that's the case, it's pretty easy to simply ignore since the analogous values (home/away_team_live_score and home/away_team_cats) are accessible from the Matchup objects,

If so, it might be a mild QoL improvement to make the Matchup class's home_team_cats/away_team_cats and home_team_live_score/away_team_live_score accessible from box_scores (in place of home_score/away_score). Removing the 'points' attribute from box_player objects in cat leagues could also help ensure clarity.

**Reply by cwendt94** [reply]
Yeah I only need the data from the ESPN box score request
```
params = {
   'view': ['mMatchupScore', 'mScoreboard'],
   'scoringPeriodId': scoring_id
 }

filters = {"schedule":{"filterMatchupPeriodIds":{"value":[matchup_id]}}}
headers = {'x-fantasy-filter': json.dumps(filters)}
data = self.espn_request.league_get(params=params, headers=headers)
schedule = data['schedule']
```

That solution could work, thanks for the input! I have never been in a category league before so I didn't realize how the scoring works.
